utils/containers/hash/hash_array.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __UTILS_CONTAINERS_HASH_HASH243_ARRAY_H__
+#define __UTILS_CONTAINERS_HASH_HASH243_ARRAY_H__
+
+#include "utarray.h"
+
+#include "common/trinary/flex_trit.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef UT_array* hash_array_p;
```

Comment:
Can we instead do `typedef UT_array* hash_array_t;` (like you've done with `typedef UT_array* transaction_array_t;`) ? Because here the base type is actually a pointer. I find `hash_array_p` to be a bit misleading because we can think we have a pointer to the type when it's not really the case.

Reply:
`UT_array` is a structure, not a pointer. so `hash_array_p` is a pointer to the UT_array structure.

Reply:
Yes, what I meant to say was that the base type we are going to handle is a `UT_array *`. It makes sense since an array IS a pointer.

Reply:
Anyway, one of `hash_array_p` or `transaction_array_t` has to change since they are contradictory, and IMHO it should be `hash_array_p`.

Reply:
IMO, `hash_array_p` and `transaction_array_t` (should be `transaction_array_p`) are two different object, hash_array is a hash container and transaction_array holds `struct _iota_transaction`.

Reply:
let's change `transaction_array_t` to `transaction_array_p`.

Reply:
They are both arrays containing different kind of objects.
As you want, but IMHO it would make more sense to have `_t` because even in C what we call an array is just a pointer so the type itself IS a pointer.

Reply:
In this case, we should change `trit_array_p` to `trit_array_t` as well, right?

Reply:
That is a good question and I actually think that this `trit_array_p` typedef shouldn't even exists (for reasons mentioned here https://github.com/iotaledger/entangled/pull/589#discussion_r238860288) and we are in a very similar situation. So why not just `typedef UT_array hash_array_t;` and use `hash_array_t *` when needed ?

Reply:
Anyway, the PR looks good to me I'll approve.
I was just sharing thoughts, I just find names like `hash_array_p` very confusing and I think we should stay away from typedefing pointers.

Reply:
It's about code readability, we agree not to use `_t` as the suffix of a structure pointer and `_p` is a reminder for me that it's a pointer, b/c I am too lazy to add `*`.  **`¯\_(⌣̯̀⌣́)_/¯`**
agree, to use typedefs of non-pointer types only.